addons/point_of_sale/models/stock_rule.py

```diff
@@ -0,0 +1,15 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+
+from odoo import models
+
+
+class StockRule(models.Model):
+    _inherit = 'stock.rule'
+
+    def _get_stock_move_values(self, product_id, product_qty, product_uom, location_id, name, origin, company_id, values):
+        move_values = super()._get_stock_move_values(product_id, product_qty, product_uom, location_id, name, origin, company_id, values)
```

Comment:
I would keep the `pos_order` flag you used in https://github.com/odoo/odoo/commit/0093126ceca7ffd482697fb497897f77bf59bdc9
to override the value only for order coming from the POS